src/renderer/widgets/RemoveVoteModal/ui/RemoveVoteModal.tsx

```diff
@@ -1,35 +1,59 @@
 import { useGate, useUnit } from 'effector-react';
+import { useEffect, useState } from 'react';
 
 import { useI18n } from '@app/providers';
-import { type AccountVote, type Asset, type Chain, type OngoingReferendum } from '@shared/core';
-import { useModalClose } from '@shared/lib/hooks';
-import { Step, isStep } from '@shared/lib/utils';
+import { type AccountVote, type Address, type Asset, type Chain, type OngoingReferendum } from '@/shared/core';
+import { useModalClose } from '@/shared/lib/hooks';
+import { Step, isStep, nonNullable, nullable } from '@/shared/lib/utils';
 import { BaseModal, Button } from '@shared/ui';
-import { OperationTitle } from '@entities/chain';
-import { OperationResult } from '@entities/transaction';
-import { OperationSign, OperationSubmit } from '@features/operations';
-import { RemoveVoteConfirmation, basketUtils } from '@features/operations/OperationsConfirm';
+import { OperationTitle } from '@/entities/chain';
+import { OperationResult } from '@/entities/transaction';
+import { OperationSign, OperationSubmit } from '@/features/operations';
+import { RemoveVoteConfirmation, basketUtils } from '@/features/operations/OperationsConfirm';
+import { SignatorySelectModal } from '@/pages/Operations/components/modals/SignatorySelectModal';
 import { removeVoteModalAggregate } from '../aggregates/removeVoteModal';
 
 type Props = {
   referendum: OngoingReferendum;
+  voter: Address;
   vote: AccountVote;
   api: ApiPromise;
   chain: Chain;
   asset: Asset;
   onClose: VoidFunction;
 };
 
-export const RemoveVoteModal = ({ referendum, vote, chain, asset, api, onClose }: Props) => {
-  useGate(removeVoteModalAggregate.gates.flow, { referendum, vote, chain, asset, api });
+export const RemoveVoteModal = ({ referendum, vote, voter, chain, asset, api, onClose }: Props) => {
+  useGate(removeVoteModalAggregate.gates.flow, { referendum, vote, voter, chain, asset, api });
 
   const { t } = useI18n();
   const step = useUnit(removeVoteModalAggregate.$step);
+  const signatory = useUnit(removeVoteModalAggregate.$signatory);
+  const signatories = useUnit(removeVoteModalAggregate.$signatories);
   const initiatorWallet = useUnit(removeVoteModalAggregate.$initiatorWallet);
 
-  const [isModalOpen, closeModal] = useModalClose(step !== Step.NONE, onClose);
-  const [isBasketModalOpen, closeBasketModal] = useModalClose(step === Step.BASKET, onClose);
+  const shouldPickSignatory = nullable(signatory) && signatories.length > 0;
+
+  const [isModalOpen, closeModal] = useModalClose(!isStep(step, Step.NONE), onClose);
+  const [isBasketModalOpen, closeBasketModal] = useModalClose(isStep(step, Step.BASKET), onClose);
+  const [isSelectSignatoryOpen, setIsSelectSignatoryOpen] = useState(shouldPickSignatory);
+  const [isSelectSignatoryClosed, setIsSelectSignatoryClosed] = useState(false);
+
+  const handleSelectSignatoryClose = () => {
+    setIsSelectSignatoryOpen(false);
+    setIsSelectSignatoryClosed(true);
+  };
```

Comment:
maybe use `useToggle(shouldPickSignatory)` instead of it?

Reply:
In this case we can't be sure, that modal closed state is caused by user interaction. `isSelectSignatoryClosed` tells effect to trigger modal close only if user closed select modal and didn't selected any signatory